### PR TITLE
#1 - Nupkg hash calculation

### PR DESCRIPTION
**Pull request by olegmoz**

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#12](https://github.com/artipie/nuget-adapter/issues/12) is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/4b21a340-4ccd-45e4-a0bd-4b63a7f37484); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by olegmoz** [comment]
@Nimmaster Please review this PR.

File: src/test/java/com/artpie/nuget/NupkgTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * Nuspec created from resource.
+     */
+    private Nupkg nupkg;
+
+    @BeforeEach
+    void init() throws Exception {
+        this.storage = new BlockingStorage(
+            new FileStorage(
+                Files.createTempDirectory(NupkgTest.class.getName()).resolve("repo")
```
Comment (Nimmaster) [inline src/test/java/com/artpie/nuget/NupkgTest.java]: Use [TempDir](https://junit.org/junit5/docs/5.4.2/api/org/junit/jupiter/api/io/TempDir.html) JUnit 5 annotation. It automatically cleans the temp directory after(Each|All) test cases.

File: src/test/java/com/artpie/nuget/NupkgTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            )
+        );
+        final byte[] bytes = new NewtonJsonResource("newtonsoft.json.12.0.3.nupkg").bytes();
```
Comment (Nimmaster) [inline src/test/java/com/artpie/nuget/NupkgTest.java]: You don't really need to put a real file in test resources. You can just generate random file in runtime right in the test case.

File: src/test/java/com/artpie/nuget/NupkgTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        this.nupkg = new Nupkg(ByteSource.wrap(bytes));
+    }
+
+    @Test
+    void shouldCalculateHash() throws Exception {
+        final PackageIdentity identity = new PackageIdentity("newtonsoft.json", "12.0.3");
+        this.nupkg.hash().save(this.storage, identity);
```
Comment (Nimmaster) [inline src/test/java/com/artpie/nuget/NupkgTest.java]: You don't need to store it in Asto too

File: src/test/java/com/artpie/nuget/NupkgTest.java (hunk 1 above)
Comment (Nimmaster) [reply src/test/java/com/artpie/nuget/NupkgTest.java]: Ah it's since v5.4. Nevermind.

Comment (olegmoz) [reply src/test/java/com/artpie/nuget/NupkgTest.java]: Thank you, changed all tests in this project to use `@TempDir`

File: src/test/java/com/artpie/nuget/NupkgTest.java (hunk 2 above)
Comment (olegmoz) [reply src/test/java/com/artpie/nuget/NupkgTest.java]: Resource removed, using example data instead

File: src/test/java/com/artpie/nuget/NupkgTest.java (hunk 3 above)
Comment (olegmoz) [reply src/test/java/com/artpie/nuget/NupkgTest.java]: Sadly I have to, saving hash to storage is the only way to see what's inside

**Comment by olegmoz** [comment]
@yegor256 Please review

File: src/test/java/com/artpie/nuget/NupkgTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * @since 0.1
+ */
+class NupkgTest {
+
+    @Test
+    void shouldCalculateHash(final @TempDir Path temp) throws Exception {
+        final BlockingStorage storage = new BlockingStorage(new FileStorage(temp));
+        final PackageIdentity identity = new PackageIdentity("foo", "1.0.0");
+        final byte[] data = "test data".getBytes(StandardCharsets.UTF_8);
```
Comment (iakunin) [inline src/test/java/com/artpie/nuget/NupkgTest.java]: @olegmoz `data` variable can be inlined.

File: src/test/java/com/artpie/nuget/NupkgTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        final Nupkg nupkg = new Nupkg(ByteSource.wrap(data));
```
Comment (iakunin) [inline src/test/java/com/artpie/nuget/NupkgTest.java]: @olegmoz also `nupkg` variable can be inlined.

**Comment by olegmoz** [comment]
@iakunin fixed

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @iakunin[/z](https://www.0crat.com/u/iakunin)[/z](https://www.0crat.com/u/iakunin) didn't make CR comments

**Comment by 0crat** [comment]
Job was finished in 21 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @iakunin[/z](https://www.0crat.com/u/iakunin)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**